### ISSUE TITLE
"{ was not closed" in f-string with escape sequence and variable width

### ISSUE THREAD

**Issue by bszonye** [microsoft/pyright#4481]
**Describe the bug**
These f-strings:
```
label = f"{i:\t>{width}s}"
label = f"{i:\x00>{width}s}"
label = f"{i:\u2007>{width}s}"
```
produce several unexpected errors:
```
xxx.py:538:33 - error: "{" was not closed
xxx.py:539:35 - error: "{" was not closed
xxx.py:540:38 - error: Expected expression
xxx.py:540:37 - error: "{" was not closed
xxx.py:538:34 - error: "widt" is not defined (reportUndefinedVariable)
xxx.py:539:36 - error: "wi" is not defined (reportUndefinedVariable)
xxx.py:539:33 - error: Operator ">" not supported for types "Literal[0]" and "set[Unknown]"
    Operator ">" not supported for types "Literal[0]" and "set[Unknown]" (reportGeneralTypeIssues)
xxx.py:540:33 - error: Operator ">" not supported for types "float" and "set[Unknown]"
    Operator ">" not supported for types "float" and "set[Unknown]" (reportGeneralTypeIssues)
8 errors, 0 warnings, 0 informations
```
The escape sequences and variable width are both fine separately. These similar lines do _not_ produce errors:
```
label = f"{i:0>{width}d}"
label = f"{i:\t>2d}"
```
Note that the error message drops a number of letters from `width` equal to the length of the escape sequence. This suggests that an index or offset is being applied to the wrong substring.

**To Reproduce**
Write an f-string literal that uses an escape sequence as a fill character and a variable parameter later in the f-string.

**Expected behavior**
The checker should parse the f-string successfully like it does with non-escaped fill characters.

**VS Code extension or command-line**
Running from command line on WSL Ubuntu:
```
$ pyright --version
pyright 1.1.289
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.290, which I just published. It will also be included in a future release of pylance.

**Comment by bszonye** [comment]
Verified, thanks!
